wallet-new/integration/Types.hs

```diff
@@ -0,0 +1,104 @@
+{-# LANGUAGE ConstraintKinds #-}
+{-# LANGUAGE DeriveGeneric   #-}
+{-# LANGUAGE StrictData      #-}
+{-# LANGUAGE TypeFamilies    #-}
+
+module Types
+    ( Probability
+    , createProbability
+    , getProbability
+    , Action (..)
+    , WalletState (..)
+    , wallets
+    , accounts
+    , addresses
+    , transactions
+    , actionsNum
+    , ActionWalletState
+    , ActionProbabilities
+    , WalletTestMode
+    ) where
+
+import           Universum
+
+import           Control.Lens (makeLenses)
+
+import           Cardano.Wallet.API.V1.Types (Account, Wallet, WalletAddress,Transaction)
+
+
+-- | Ideally, we would put @MonadGen@ here and remove @MonadIO@,
+-- but it's better to see how the client fits in the end.
+type WalletTestMode m =
+    ( MonadIO m
+    , MonadThrow m
+    , MonadPlus m
+    )
+
+-- | The probability type that captures the chance of
+-- our random action selection choosing a specific @Action@.
+-- The number is restricted to 1 - 100 - see @createProbability@.
+data Probability = Probability { getProbability :: Int }
+    deriving (Show, Eq)
+
+
+-- | Safe constructor that checks the values.
+createProbability :: Int -> Probability
+createProbability prob
+    | prob <= 0 || prob > 100 = error "Invalid range. 0 - 100."
```

Comment:
Regarding this, I do have something to comment. I really don't like the idea of using `Maybe` in places where we can have an error. 
In this scenario, it's much more natural to use `Either` and I used it. Running a function and getting `Nothing` pisses me off since I have to look at "Why?", while `Either` can immediately tell you what's the issue.
All that aside, I don't really see the benefit of using some additional constructs besides `error` here since the end result in this case will be the same - the user will see an error pop up and the error will be "Wrong probability ...". Sometimes less is more, and I think this is one of these cases. I know we are all pure and shiny, but if I have a choice to do something more complicated with no change in code semantics, I don't see the point :hear_no_evil: